src/frontend/mame/luaengine.cpp

```diff
@@ -987,6 +987,38 @@ void lua_engine::initialize()
 			return table;
 		}));
 
+/*  emu_options library
+ *
+ * manager:options()
+ * manager:machine():options()
+ *
+ * options:slot_option() - retrieves a specific slot option
+ *
+ * options.entries[] - get table of option entries (k=name, v=core_options::entry)
+ */
+
+	auto emu_options_type = sol().registry().new_usertype<emu_options>("new", sol::no_constructor, sol::base_classes, sol::bases<core_options>());
+	emu_options_type.set("slot_option", [](emu_options &opts, const std::string &name) { return opts.find_slot_option(name); });
+	sol().registry().set("emu_options", emu_options_type);
+
+/*  slot_option library
+ *
+ * manager:options():slot_option("name")
+ * manager:machine():options():slot_option("name")
+ *
+ * slot_option:value()
+ * slot_option:specified_value()
+ * slot_option:bios()
+ * slot_option:default_card_software()
```

Comment:
At least some of these would probably work better as read-only properties.  Also, try to give a description for at least some of them.